core/src/main/java/org/apache/calcite/adapter/enumerable/EnumUtils.java

```diff
@@ -268,6 +276,48 @@ static JoinType toLinq4jJoinType(JoinRelType joinRelType) {
     throw new IllegalStateException(
         "Unable to convert " + joinRelType + " to Linq4j JoinType");
   }
+
+  /**
+   * Generate Expression for Join condition, 'excludeEquiCondition'
+   * indicates whether to exclude equi-join conditions for the result.
+   */
+  static Expression generatePredicateForJoin(
```

Comment:
@jinxing64 please note that this refactoring was already implemented via https://github.com/apache/calcite/pull/1424/

Reply:
Yes, thanks ruben~~
Actually, I'm reverting at the moment :)